collection-scripts/gather

```diff
@@ -12,8 +15,11 @@ resources+=(clusteroperators)
 # Certificate Resources
 resources+=(certificatesigningrequests)
 
-# Machine/Node Resources
-resources+=(nodes machines machineconfigs machineconfigpools)
+# MCO/Node Resources
+resources+=(nodes machineconfigs machineconfigpools)
```

Comment:
Do we have a relation ship between these? 
IE: If we get a node; do we need to know its machineconfigs or machineconfigpools? 

If so - it might be better to put this relationship into to the must-gather binary so every time a node is collected *(or one of the other objects); the proper related objects are collected.

Reply:
I can't speak for everyone though I believe `machineconfigs` and `machineconfigpools` are namespace-less. That's why we need to query for machines separately since they live only under `openshift-machine-api` namespace.

Reply:
@runcom FYI ^^

Reply:
See https://github.com/openshift/machine-config-operator/blob/master/docs/MachineConfigServer.md#detailed-design

Reply:
@sferich888 I have a feeling it's impossible to know which machine config was used for a given node. Based on posted [1], I don't think node object has any field/label/annotation that can say which endpoint was used. Also, worker nodes have different image in general. Each image knows an endpoint (most likely it's hard-coded in it when it's baked). Due to security reason so you can't pretend to be master when you are worker. Or, I am wrong @runcom @ashcrow ?

[1] https://github.com/openshift/machine-config-operator/blob/master/docs/MachineConfigServer.md#detailed-design